### PR TITLE
Remove client side query parsing

### PR DESCRIPTION
**Pull request by jd**
This removes the client side parsing of query that was supported and sends it
directly to the server. This will required Gnocchi 3.0 at least to be used.

Sem-Ver: api-break

### CODE REVIEW

File: gnocchiclient/v1/metric.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def get_measures(self, metric, start=None, stop=None, aggregation=None,
         return [(iso8601.parse_date(ts), g, value)
                 for ts, g, value in measures]
 
-    def aggregation(self, metrics, query=None,
+    def aggregation(self, metrics, query=None, filter=None,
```
Comment (sileht) [inline gnocchiclient/v1/metric.py]: We can have only one param and check if it's a string or a dict ?

Comment (jd) [reply gnocchiclient/v1/metric.py]: Good idea, let's do that.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 0b341fd6326f626c2d79ce54b45d8939c7bd37f9

**Comment by jd** [comment]
@chungg good question. That really depends how you want to do that, because it is only some features not working while other works since 1.0. In this case it's just the search/filter from the CLI with text that will be broken with Gnocchi < 3. If you don't use the CLI nor Gnocchi < 3, or never search, nothing's broken :)

It's not like it'd be a major API change or anything.

**Comment by chungg** [comment]
i'm pretty sure as a user, compatible == nothing broken.

**Comment by jd** [comment]
I did not say it was entirely compatible, I'm just saying it's not that easy to track. The commit has `Sem-Ver: api-break` which makes us (in theory) release a new major version (so client v5 or something like that) which indicates compat is broken :)

**Comment by jd** [comment]
Pull-request updated, HEAD is now b1275d9dba35c68ab0daf9a8b898fe234c78a1e7

**Comment by jd** [comment]
Pull-request updated, HEAD is now 89d1bc0de60c944d68fd1d890c645243f9082665

**Comment by jd** [comment]
Pull-request updated, HEAD is now cb2c56d5c482c3c5040c826d10af0784a621f517

**Comment by jd** [comment]
Pull-request updated, HEAD is now df35a0ec10ef91348cfe874f7a88964360db1900

**Comment by jd** [comment]
Pull-request updated, HEAD is now 2d632610909b40760418bc5968c9c4b7f286f595

**Comment by jd** [comment]
Pull-request updated, HEAD is now 2e1180d4728e31010065f0a9eaac40750534f667

**Comment by jd** [comment]
Pull-request updated, HEAD is now 8ac22176d899af210778c5d744f16b7ae06c0ccc

**Comment by jd** [comment]
Pull-request updated, HEAD is now 409789cd4c22ebb1509bfbc4da1338fc95b4e683

**Pull request merged**